File: AlphaMedic/Rest/Models/Appointment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Rest.Models
{
	public enum AppointmentState
	{
		Unconfirmed,
		Accepted,
		Finished
	}

	public class Appointment
	{
		[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		[Key]
		public int AppointmentId { get; set; }

		public AppointmentState State { get; set; }

		[ForeignKey("Doctor")]
		public int? DoctorId { get; set; }

		[ForeignKey("Patient")]
		public int PatientId { get; set; }

		public string Description { get; set; }

		public DateTime? Date { get; set; }

		[DataType(DataType.Time)]
		public TimeSpan? Duration { get; set; }

		public bool? Checked { get; set; }

		public virtual Doctor Doctor { get; set; }
		public virtual Procedure Procedure { get; set; }
		public virtual Patient Patient { get; set; }

	}
}

File: AlphaMedicAndroidApplication/app/src/main/java/com/example/oleksandrkliushta/testapplication/model/Appointment.java
package com.example.oleksandrkliushta.testapplication.model;

import java.sql.Time;
import java.util.Date;


/**
 * Created by oleksandr.kliushta on 9/8/2016.
 */
public class Appointment
{
    public int AppointmentId;
    public AppointmentState State;
    public int DoctorId;
    public int PatientId;
    public String Surname;
    public String Name;
    public String Description;

    public String Date;
    public String Duration;

    public Appointment() {}

    public Appointment(String dateTime, String description) {
        this.Date = dateTime;
        this.Description = description;
        this.State = AppointmentState.Unconfirmed;
    }

    public enum AppointmentState
    {
        Unconfirmed,
        Accepted,
    }
}



